src/maplibre_helpers.ts

```diff
@@ -229,18 +229,26 @@ export function setPrecision(pt: Position): Position {
 //  Helper for https://maplibre.org/maplibre-style-spec/expressions/#match.
 //  Gets one feature property, uses a map to match a key to a value, and
 //  includes a fallback if no keys match.
-export function matchValue<Output>(
+export function getDataDrivenPropertyValueSpecification<OutputType>(
```

Comment:
Hmm, not sure this name makes sense to me. DataDrivenPropertyValueSpecification is described (not with this name) at https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#data-expressions. It's a general way of describing things in maplibre styles that get calculated per feature. This method is much more specific -- it's just for `match`, linked above. It's basically a switch-case statement:

1) Fetch some property of the feature (getter)
2) Go through each of the key/value pairs in the mapping and try to match
3) If that fails, use a fallback value

If `matchValue` is too terse, how about something like `pickValueBasedOnKey` or something like that?

Reply:
The thing is, it's not quite `pickValueBasedOnKey`, because you're getting an object from which you can pick a value right?

So my main issue with the name is that you're returning a data structure rather than the end matched result, to my understanding. I agree this name isn't great either though

Reply:
Something like `getPropertyMatcher` or `constructMatchExpression`?

Reply:
> you're returning a data structure rather than the end matched result

That's right, and it's how style expressions work in general. We're describing to maplibre how to choose a color, opacity, line width, etc.

Both those suggestions sound good to me, `constructMatchExpression` would get my vote